.github/workflows/builds-and-tests.yml

```
@@ -101,181 +101,189 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-    # This external action allows to interrupt a workflow already running on
-    # the same branch to save resource
-    - name: Cancel Previous Runs
-      uses: styfle/cancel-workflow-action@0.11.0
-      with:
-        access_token: ${{ github.token }}
+      # This external action allows to interrupt a workflow already running on
+      # the same branch to save resources.
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          access_token: ${{ github.token }}
 
-    # Checkout MFEM in "mfem" subdirectory. Final path:
-    # /home/runner/work/mfem/mfem/mfem
-    # Note: Done now to access "install-hypre" and "install-metis" actions.
-    - name: checkout mfem
-      uses: actions/checkout@v3
-      with:
-        path: ${{ env.MFEM_TOP_DIR }}
-        # Fetch the complete history for codecov to access commits ID
-        fetch-depth: 0
+      # Fix 'No space left on device' errors for Ubuntu builds.
+      - name: Run Actions Cleaner
+        if: matrix.os == 'ubuntu-latest'
+        uses: easimon/maximize-build-space@master
```

Maybe we want to specify a version rather than master to avoid issues if they make breaking changes?

```suggestion
        uses: easimon/maximize-build-space@v8
```

---

👍🏻 
This is also a best practice security-wise.

---

Done in [b8fb0fa](https://github.com/mfem/mfem/pull/3902/commits/b8fb0faba9f5f9d0574141b98ca642b9acb38730).